src/adapter/src/catalog.rs

```diff
@@ -472,13 +472,19 @@ impl CatalogState {
             .flatten()
     }
 
+    /// Associates a name, `GlobalId`, and entry. Returns the previous
+    /// `GlobalId` associated with that name, if any.
+    ///
+    /// The returned value is useful only in the case of builtin types, which
+    /// do not otherwise go through name resolution to ensure the desired name
+    /// is not occupied.
```

Comment:
Are you ever meant to do anything with the returned global ID besides assert that it is not none? ISTM that assertion should be pushed into this function, so that each caller doesn't need to perform the assertion independently!

Reply:
Oh--good call! I had assumed because this check wasn't in place that this function was meant to handle updates or be idempotent, but from the looks of things that is not the case.